prov/efa/src/efa_mr.c

```diff
@@ -551,8 +552,11 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 		*/
 		original_access = mr_attr->access;
 		mr_attr->access |= FI_REMOTE_READ;
+		shm_flags = flags;
+		if (mr_attr->iface != FI_HMEM_SYSTEM)
```

Comment:
Could you add a comment here to call it out as well?

Reply:
added